# Урок 2. Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона

In [1]:
from math import factorial
import numpy as np

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [3]:
def arrangements(n, k):
    return int(factorial(n) / factorial(n - k))

In [4]:
def permutations(n):
    return int(factorial(n))

## 1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Искомая вероятность находится по формуле Бернулли:

## $$P_n^k= C_n^kp^kq^{n-k}$$

**где** $q=1-p$.

В текущем случае $n = 100$, $k=85$, $p=0,8$, $q=0,2$. В итоге имеем:

## $$P_{100}^{85}= C_{100}^{85}\cdot0,8^{85}\cdot0,2^{100-85}$$

In [5]:
combinations(100, 85)*0.8**85*0.2**15

0.048061793700746556

### __Ответ__: 0.048061793700746556

## 2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

Так как вероятность события мала, и количество испытаний велико, применим формулу Пуассона:

## $$P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$$

**где** $\lambda=np$.

В текущем случае $n = 5000$, $p=0,0004$, $\lambda=np=5000\cdot{0,0004}=2$.

Вероятность, что ни одна из них не перегорит в первый день:

## $$P_0 \approx \frac{2^0}{0!}e^{-2}$$

In [6]:
(2**0/factorial(0))*np.exp(-2)

0.1353352832366127

Вероятность, что перегорят ровно две:

## $$P_2 \approx \frac{2^2}{2!}e^{-2}$$

In [7]:
(2**2/factorial(2))*np.exp(-2)

0.2706705664732254

### __Ответ__: а) 0.1353352832366127; б) 0.2706705664732254

## 3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Искомая вероятность находится по формуле Бернулли:

## $$P_n^k= C_n^kp^kq^{n-k}$$

**где** $q=1-p$.

В текущем случае $n = 144$, $k=70$, $p=0,5$, $q=0,5$ (выпадание ребра не рассматриваем). В итоге имеем:

## $$P_{144}^{70}= C_{144}^{70}\cdot0,5^{70}\cdot0,5^{144-70}$$

In [8]:
combinations(144, 70)*(0.5**70)*(0.5**74)

0.06281178035144776

### __Ответ__: 0.06281178035144776

## 4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

#### 1)
Вероятность того, что из первого ящика будет вытащено 2 белых мяча:

In [9]:
(7/10)*(6/9)

0.4666666666666666

Вероятность того, что из второго ящика будет вытащено 2 белых мяча:

In [10]:
(9/11)*(8/10)

0.6545454545454547

Вероятность того, что все мячи белые:

In [11]:
(7/10)*(6/9)*(9/11)*(8/10)

0.3054545454545454

#### 2)
Вероятность того, что ровно 2 мяча белые равна сумме вероятностей событий:

а) 2 мяча из первого ящика белые и 2 мяча из второго - чёрные;

б) 1 мяч из первого ящика белый и 1 мяч из первого ящика чёрный и 1 мяч из второго ящика белый и 1 мяч из второго ящика чёрный;

в) 2 мяча из первого ящика чёрные и 2 мяча из второго - белые.

##### а)

Вероятность того, что из второго ящика будет вытащено 2 чёрных мяча:

In [12]:
(2/11)*(1/10)

0.018181818181818184

Вероятность того, что 2 мяча из первого ящика белые и 2 мяча из второго - чёрные:

In [13]:
(7/10)*(6/9)*(2/11)*(1/10)

0.008484848484848484

##### б)

Вероятность того, что 1 мяч из первого ящика белый и 1 мяч из первого ящика чёрный:

In [14]:
# Порядок не важен, поэтому благоприятные 2 исхода:
# - Cначала белый потом чёрный;
# - Сначала чёрный потом белый.
(7/10)*(3/9) + (3/10)*(7/9)

0.4666666666666667

Вероятность того, что 1 мяч из второго ящика белый и 1 мяч из второго ящика чёрный:

In [15]:
# Порядок не важен, поэтому благоприятные 2 исхода:
# - Cначала белый потом чёрный;
# - Сначала чёрный потом белый.
(9/11)*(2/10) + (2/11)*(9/10)

0.32727272727272727

Вероятность того, что 1 мяч из первого ящика белый и 1 мяч из первого ящика чёрный и 1 мяч из второго ящика белый и 1 мяч из второго ящика чёрный:

In [16]:
((7/10)*(3/9) + (3/10)*(7/9))*((9/11)*(2/10) + (2/11)*(9/10))

0.15272727272727274

##### в)

Вероятность того, что из первого ящика будет вытащено 2 чёрных мяча:

In [17]:
(3/10)*(2/9)

0.06666666666666667

Вероятность того, что 2 мяча из первого ящика чёрные и 2 мяча из второго - белые:

In [18]:
(3/10)*(2/9)*(9/11)*(8/10)

0.04363636363636364

Вероятность того, что ровно два мяча белые:

In [19]:
(7/10)*(6/9)*(2/11)*(1/10) + ((7/10)*(3/9) + (3/10)*(7/9))*((9/11)*(2/10) + (2/11)*(9/10)) + (3/10)*(2/9)*(9/11)*(8/10)

0.20484848484848486

#### 3)
Найдём вероятность того, что все мячи чёрные:

In [20]:
(3/10)*(2/9)*(2/11)*(1/10)

0.0012121212121212121

Так как это событие противополножное, то искомая вероятность равна:

In [21]:
1 - (3/10)*(2/9)*(2/11)*(1/10)

0.9987878787878788

### __Ответ__: 1) 0.3054545454545454; 2) 0.20484848484848486; 3) 0.9987878787878788